internal/database/postgres/postgres.go

```diff
@@ -148,57 +153,23 @@ func updateLastSeen(host string, now time.Time, tx *sql.Tx) (err error) {
 	return
 }
 
-func (conn postgresConn) AppendDataPoint(host string, packet uplink.GPUStatSample) error {
-	var err error
-
-	tx, err := conn.db.Begin()
-	if err != nil {
-		return err
-	}
-
-	// Find matching GPU
-
-	// TODO: replace with Query
-	// This silently discards all rows other than the first
-	row := tx.QueryRow(`SELECT Id
-		FROM GPUs
-		WHERE Machine=$1`,
-		host)
-	var id int64
-	err = row.Scan(&id)
-
-	// check error type. If no rows found, add a new GPU
-	if err != nil {
-		if errors.Is(err, sql.ErrNoRows) {
-			// Didn't find an existing entry, so add a new GPU
-
-			// use TOFU approach, only set the name, brand, etc. on
-			// the first packet; future AppendDataPoint calls only
-			// add the 'dynamic' data to Stats
-			// TODO: reevaluate whether this is a good idea (what
-			// happens when a driver update occurs?)
-
-			// TODO: the great refactor made this code impossible. Commented out for now because it wont work
-			/*
-				id, err = createGPU(host, packet, tx)
-				if err != nil {
-					return errors.Join(err, tx.Rollback())
-				}
-			*/
-			return errors.Join(err, tx.Rollback())
-		} else {
-			return errors.Join(err, tx.Rollback())
-		}
-	}
-
-	// GPU should now exist, so push stats
+func (conn postgresConn) AppendDataPoint(sample uplink.GPUStatSample) error {
 	now := time.Now()
-	err = insertStats(id, packet, now, tx)
-	if err != nil {
-		return errors.Join(err, tx.Rollback())
-	}
 
-	return tx.Commit()
+	_, err := conn.db.Exec(`INSERT INTO Stats
+		(Gpu, Recieved, MemoryUtilisation, GpuUtilisation, MemoryUsed,
```

Comment:
can we spell received properly now hehe

Reply:
Damn it. I swear I passed GCSE english. Will fix in a bit